libunicode.c

```diff
@@ -107,11 +112,16 @@ static int lre_case_conv_entry(uint32_t *res, uint32_t c, int conv_type, uint32_
         c = case_conv_ext[data] + (conv_type == 2);
         break;
     case RUN_TYPE_U_EXT:
+    case RUN_TYPE_L_EXT:
     case RUN_TYPE_LF_EXT:
         if (is_lower != (type - RUN_TYPE_U_EXT))
             break;
         c = case_conv_ext[data];
         break;
+    case RUN_TYPE_U_EXT2:
+    case RUN_TYPE_U_EXT3:
+        // TODO
```

Comment:
Someone needs to tell me how to implement these, my brain is not capable of writing this kind of obfuscated C code.

Reply:
Caveat emptor, I'm not 100% sure but I suspect you can merge the `RUN_TYPE_U_EXT2` case with `RUN_TYPE_LF_EXT2` and update the guard to:
```c
if (is_lower != (type - RUN_TYPE_U_EXT2))
    break;
```
`RUN_TYPE_U_EXT2` should immediately precede `RUN_TYPE_LF_EXT2` in the enum declaration for that to work (not the case right now.)

For `RUN_TYPE_U_EXT3`, the code should probably look like this:
```c
case RUN_TYPE_U_EXT3:
    res[0] = c - code + case_conv_ext[data >> 8];
    res[1] = case_conv_ext[(data >> 4) & 0x0f];
    res[2] = case_conv_ext[data & 0x0f];
    return 3;
```